controllers/student.js

```diff
@@ -0,0 +1,41 @@
+const bcrypt = require('bcryptjs');
+const BigPromise = require('../middlewares/BigPromise.js');
+const User = require('../models/userSchema.js');
+
+const registerStudent = BigPromise(async (req, res, next) => {
+  // TODO : What to do with Contact?
+  let { name, email, department, role, id, password, confirmPassword } = req.body;
```

Comment:
You missed **roll_number**. Also id will not come from frontend. We will generate a unique id on backend for every student and while storing the data in Mongo DB we will use that id and map it to a student.

Confirm password is also not needed, since we use it on frontend to double check if student has correctly written the password or not.